modules/spechla/1.0/spechla.smk

```diff
@@ -0,0 +1,310 @@
+        chmod +x -R {params.scriptdir}/bin/*
+        """)
+
+rule _spechla_index:
+    input:
+        txt = CFG["dirs"]["inputs"] + "SpecHLA/clone.done"
+    output:
+        txt = touch(CFG["dirs"]["inputs"] + "SpecHLA/index.done")
+    log:
+        stdout = CFG["logs"]["inputs"] + "index.stdout.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA"
+    threads:
+        CFG["threads"]["setup"]
+    resources:
+        **CFG["resources"]["setup"]  
+    shell:
+        op.as_one_line("""
+        unset LD_LIBRARY_PATH && unset LIBRARY_PATH &&
+        bash {params.scriptdir}/index.sh > {log.stdout} 2>&1
+        """)
+
+# Extract HLA reads from bam and cram
+def _which_genome(wildcards):
+    if "38" in str({wildcards.genome_build}):
+        return "hg38"
+    else:
+        return "hg19"
+
+rule _spechla_extract_reads:
+    input:
+        txt = CFG["dirs"]["inputs"] + "SpecHLA/index.done",
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai",
+        crai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.crai"
+    output:
+        fastq_1 = temp(CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_1.fq.gz"),
+        fastq_2 = temp(CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_2.fq.gz"),
+        fastq_unpaired = temp(CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract.unpaired.fq.gz"),
+    log:
+        stderr = CFG["logs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}/extract_reads.stderr.log"
+    params:
+        scriptdir = CFG["dirs"]["inputs"] + "SpecHLA",
+        sample_id = "{sample_id}",
+        genome = _which_genome,
+        outdir = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/"
+    conda: CFG["conda_envs"]["spechla"]
+    group: "extract_and_run"
+    threads:
+        CFG["threads"]["extract_reads"]
+    resources:
+        **CFG["resources"]["extract_reads"]  
+    shell:
+        op.as_one_line("""
+        bash {params.scriptdir}/script/ExtractHLAread.sh -s {params.sample_id} -b {input.bam} -r {params.genome} -o {params.outdir} > {log.stderr} 2>&1
+        """)
+
+#Choose full-length or exon typing [0|1]. 0 indicates full-length, 1 means exon,
+#default is 0. With Exome or RNA data, must select 1 (i.e., exon typing).
+def _which_seqtype(wildcards):
+    if any(seq_type in str({wildcards.seq_type}) for seq_type in ['exome', 'capture', 'target', 'mrna']):
+        return 1
+    else:
+        return 0
+
+rule _spechla_hla_typing:
+    input:
+        fq_1 = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_1.fq.gz",
+        fq_2 = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract_2.fq.gz",fastq_unpaired = CFG["dirs"]["hla_reads"] + "{seq_type}--{genome_build}/{sample_id}_extract.unpaired.fq.gz"
```

Comment:
This line seems to be missing a return (I mean this should be 2 lines instead of 1)